tip.lic

```diff
@@ -0,0 +1,99 @@
+# someone, in a similar syntax, but use the denomination to do so (e.g. "platinum").
+# <;tip gildaren 10 plat kron> now works. If you don't specify a currency,
+# this script also chooses the currency of the greatest coin you have on hand.
+
+custom_require.call %w[common common-money]
+
+class Tip
+
+  # include DRC for the message function
+  # include DRCM for use of converting amounts to copper
+  
+  def initialize
+    arg_definitions = [
+      [
+        { name: 'recipient', regex: /^[A-z\.\s\-']+$/i, variable: true, description: 'The recipent of your tip. Abbreviations ok.' },
+        { name: 'amount', regex: /^\d+$/i, variable: true, description: 'Numeric amount to tip, no commas (e.g. the number "100" in 100 kronars)' },
+        { name: 'denomination', regex: /(\bp(l|la|lat|lati|latin|latinu|latinum)?\b)|(\bg(o|ol|old)?\b)|(\bs(i|il|ilv|ilve|ilver)?\b)|(\bb(r|ro|ron|ronz|ronze)?\b)|(\bc(o|op|opp|oppe|opper)?\b)/i, variable: true, description: 'The denomination. Abbrevations ok. (e.g. plat, gold, silv)' },
+        { name: 'currency', regex: /(\bk(r|ro|ron|rona|ronar|ronars)?\b)|(l(i|ir|iru|irum|irums)?\b)|(\bd(o|ok|oko|okor|okora|okoras)?\b)/i, optional: true, variable: true, description: 'Optional currency. Abbreviations ok: kron, lirums, dokor' }         
+      ]
+    ]
+    args = parse_args(arg_definitions)  
+    gratuity(args.recipient, args.amount, args.denomination, args.currency)
+
+  end
+
+  def gratuity(recipient, amount, denomination, currency)
+    
+    total = convert_total(amount, denomination)
+    currency = handle_currency(total, currency)
+    
+    fput("tip #{recipient} #{total} #{currency}")
+    
+  end
+
+  def convert_total(amount, denomination)
+    # Take the amount, and the denomination, and determine the total
+    # in coppers. For example, 1 platinum would be 10000 coppers.
+
+    full_denom = nil
+
+    case denomination
+    when /\bp(l|la|lat|lati|latin|latinu|latinum)?\b/i
+      full_denom = "platinum"
+      return DRCM.convert_to_copper(amount, full_denom)
+    when /\bg(o|ol|old)?\b/i
+      full_denom = "gold"
+      return DRCM.convert_to_copper(amount, full_denom)
+    when /\bs(i|il|ilv|ilve|ilver)?\b/i
+      full_denom = "silver"
+      return DRCM.convert_to_copper(amount, full_denom)
+    when /\bb(r|ro|ron|ronz|ronze)?\b/i
+      full_denom = "bronze"
+      return DRCM.convert_to_copper(amount, full_denom)
+    when /\bc(o|op|opp|oppe|opper)?\b/i
+      full_denom = "copper"
+      return DRCM.convert_to_copper(amount, full_denom)
+    end
+  end
+
+  def handle_currency(total, currency)
+    # If user hasn't specified the currency, choose based
+    # on max currency on hand, else parse specified currency
+    # in case user has abbreviated it.
+    unless currency
+      # Get current wealth by currency
+      purse = {
+        "kronars": DRCM.check_wealth("Kronars"),
+        "lirums":  DRCM.check_wealth("Lirums"),
+        "dokoras": DRCM.check_wealth("dokoras")
+      }
```

Comment:
💡 Not necessary for this pull request, but to avoid multiple calls to `wealth <currency>` game command, perhaps the `DRCM.wealth` function could be updated such that if no currency argument is provided then do `wealth` and parse the output to infer each currency.

```
> wealth

Wealth:
  3 gold Kronars (3000 copper Kronars).
  9 bronze, and 2 copper Lirums (92 copper Lirums).
  No Dokoras.
```

Would return the `purse` variable you've defined.
```ruby
{
  "kronars" => 3000,
  "lirums" => 92,
  "dokoras" => 0
}

Reply:
Another good idea. If I make this changed it'll be fully backward compatible, right?

Reply:
After reviewing the `DRCM.check_wealth(currency)` and `DRCM.wealth(hometown)` methods, it may be more prudent to introduce a new function `DRCM.get_wealth` that takes no arguments and returns the hash of your wealth broken down by currency.

Then the `DRCM.check_wealth(currency)` method can be refactored to call `DRCM.get_wealth[currency]`, or be left as-is if you really didn't want to touch existing code. That function "ain't broke", my goal is to have a way that the tip script didn't need to spam `wealth` command 3 times :)